jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java

```diff
@@ -1253,4 +1253,51 @@ public void testHandleErrorOnNotSuccessfulResponseHandlingException() {
             assertEquals("COMPLETE", ex.getStrategy().name());
         }
     }
+    
+    @Test
+    public void testCookieSingleCookie() {
+        RESTWorkItemHandler handler = new RESTWorkItemHandler();
+        String headerKey = "cookieParam1";
+        String headerValue = "cookieParam1_Value";
+        String headers = headerKey + "=" + headerValue;
+
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setParameter("Url",
+                              serverURL + "/testSingleCookie");
+        workItem.setParameter("Method",
+                              "GET");
+        workItem.setParameter("Cookies", headers);
+
+        WorkItemManager manager = new TestWorkItemManager();
+        handler.executeWorkItem(workItem,
+                                manager);
+        Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+        String result = (String) results.get(PARAM_RESULT);
+        assertEquals(result, "cookieParam1:cookieParam1_Value");
+    }
+
+    @Test
+    public void testCookieMultipleCookie() {
+        RESTWorkItemHandler handler = new RESTWorkItemHandler();
+        String cookieName1 = "cookieParam1";
+        String cookieValue1 = "cookieParam1_Value";
+        String cookieName2 = "cookieParam2";
+        String cookieValue2 = "cookieParam2_Value";
+        String cookie = cookieName1 + "=" + cookieValue1 + ";" + cookieName2 + "=" + cookieValue2;
+
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setParameter("Url",
+                               serverURL + "/testMultipleCookie");
+        workItem.setParameter("Method",
+                              "GET");
+        workItem.setParameter("Cookies", cookie);
+
+        WorkItemManager manager = new TestWorkItemManager();
+        handler.executeWorkItem(workItem,
+                                manager);
+        Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+        String result = (String) results.get(PARAM_RESULT);
+        assertEquals(result, "cookieParam1:cookieParam1_Value:cookieParam2:cookieParam2_Value");
+    }
+
```

Comment:
Add test for rainy scenarios:
- malformed cookies: wrong separators, incorrect format
- null, empty cookies

Reply:
Added additional tests